### PR TITLE
Fix NcSignalPath and improvements

### PR DESCRIPTION
**Pull request by cristian-recoseanu**
- Fix the NcSignalPath implementation
- Improve Class discovery
- Improve Datatype discovery

### CODE REVIEW

File: docs/idl/NC-Framework.webidl [hunk 1]
```diff
@@ -282,44 +284,63 @@ $macro(ModelDatatypes)
...
 	enum NcDatatypeType {
-		"primitive",	// 0 primitive, e.g. NcUint16
-		"typedef",		// 1 typedef, i,e. simple alias of another datatype
-		"struct",		// 2 data structure
-		"enum",			// 3 enumeration
-		"null"			// 4 null
+		"Primitive",	// 0 primitive, e.g. NcUint16
+		"Typedef",		// 1 typedef, i,e. simple alias of another datatype
+		"Struct",		// 2 data structure
+		"Enum"
```
Comment (JeffBerryman) [inline docs/idl/NC-Framework.webidl]: I think the Enum option needs the comment "// 3 enum".

File: docs/idl/NC-Framework.webidl [hunk 2]
```diff
@@ -282,44 +284,63 @@ $macro(ModelDatatypes)
...
-		NcBoolean		persistent;	// TRUE iff property value survives power-on reset
-		NcBoolean		required;	// TRUE iff property must be implemented
+		NcPropertyId	id;				// element ID of property
+		NcName			name;			// name of property
+		NcName			typeName;		// name of property's datatype. If empty then the type is any
+		NcBoolean		readOnly;		// TRUE iff property is read-only
+		NcBoolean		persistent;		// TRUE iff property value survives power-on reset
+		NcBoolean		required;		// TRUE iff property must be implemented
+		NcBoolean		isCollection;	// TRUE iff property is a collection
 	};
```
Comment (JeffBerryman) [inline docs/idl/NC-Framework.webidl:336]: For collection properties, I wonder if we need a field that gives the dimensionality.  For example, we'll need a 2D list datatype to describe the members of a matrix.

File: docs/idl/NC-Framework.webidl [hunk 3]
```diff
@@ -282,44 +284,63 @@ $macro(ModelDatatypes)
...
 	
 	// Descriptor of a field of a struct
 	interface NcFieldDescriptor {
-		NcName	name;		// name of field
-		NcName	typeName;	// name of field's datatype
+		NcName		name;			// name of field
+		NcName		typeName;		// name of field's datatype
+		NcBoolean	isNullable;		// TRUE iff the field is nullable
+		NcBoolean	isCollection;	// TRUE iff the field is a collection
 	};
```
Comment (JeffBerryman) [inline docs/idl/NC-Framework.webidl:344]: See my comment on collections for 'properties', above.

**Review by JeffBerryman — APPROVED** [review]

**Comment by cristian-recoseanu** [comment]
Approved in the meeting on 03/05/2022

**Pull request merged**